src/packet/header/standard_header.rs

```diff
@@ -1,37 +1,78 @@
 use super::{HeaderReader, HeaderWriter};
 use crate::error::Result;
-use crate::infrastructure::DeliveryMethod;
 use crate::net::constants::STANDARD_HEADER_SIZE;
-use crate::packet::PacketTypeId;
+use crate::packet::{DeliveryGuarantee, EnumConverter, OrderingGuarantee, PacketType};
 use crate::protocol_version::ProtocolVersion;
 use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
 use std::io::Cursor;
 
 #[derive(Copy, Clone, Debug)]
 /// This header will be included in each packet, and contains some basic information.
 pub struct StandardHeader {
-    /// crc32 of the protocol version.
-    pub protocol_version: u32,
-    /// specifies the packet type.
-    pub packet_type_id: PacketTypeId,
-    /// specifies how this packet should be processed.
-    pub delivery_method: DeliveryMethod,
+    protocol_version: u32,
+    packet_type: PacketType,
+    delivery_guarantee: DeliveryGuarantee,
+    ordering_guarantee: OrderingGuarantee,
 }
 
 impl StandardHeader {
     /// Create new heartbeat header.
-    pub fn new(delivery_method: DeliveryMethod, packet_type_id: PacketTypeId) -> Self {
+    pub fn new(
+        delivery_guarantee: DeliveryGuarantee,
+        ordering_guarantee: OrderingGuarantee,
+        packet_type: PacketType,
+    ) -> Self {
         StandardHeader {
             protocol_version: ProtocolVersion::get_crc32(),
-            packet_type_id,
-            delivery_method,
+            packet_type,
+            delivery_guarantee,
+            ordering_guarantee,
         }
     }
+
+    /// Returns the protocol version
+    #[cfg(test)]
+    pub fn protocol_version(&self) -> u32 {
+        self.protocol_version
+    }
+
+    /// Returns the DeliveryGuarantee
+    pub fn delivery_guarantee(&self) -> DeliveryGuarantee {
+        self.delivery_guarantee
+    }
+
+    /// Returns the OrderingGuarantee
+    pub fn ordering_guarantee(&self) -> OrderingGuarantee {
+        self.ordering_guarantee
+    }
+
+    /// Returns the PacketType
+    #[cfg(test)]
+    pub fn packet_type(&self) -> PacketType {
+        self.packet_type
+    }
+
+    /// Returns true if the packet is a fragment, false if not
+    pub fn is_fragment(&self) -> bool {
+        self.packet_type == PacketType::Fragment
+    }
+
+    /// Checks if the protocol version in the packet is a valid version
+    pub fn is_current_protocol(&self) -> bool {
+        if !ProtocolVersion::valid_version(self.protocol_version) {
```

Comment:
Big pet peeve of mine. This should just be `ProtocolVersion::valid_version(self.protocol_version)`

Reply:
This is actually really stupid of me, lol, thanks for highlighting this.